### PR TITLE
Fix parsing tag and fields with missing name or values

### LINKED ISSUE #2678: Server allows tags with an empty string for the key and/or value
Built influxdb from source at 22542ad05bea0ce2a82cbf9bbd97aaf3e0a2c6f3.

Server allows tags with an empty string for the key and/or value, but should disallow those cases.

```
$ curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE foo"
{"results":[{}]}

$ curl -d '{"database":"foo", "points": [{"measurement": "cpu","fields": {"value": 100}, "tags": {"tag_with_empty_value":""} }]}' -H "Content-Type: application/json" http://localhost:8086/write

$ curl -G http://localhost:8086/query --data-urlencode "q=SHOW SERIES" --data-urlencode "db=foo"
{"results":[{"series":[{"name":"cpu","columns":["_id","tag_with_empty_value"],"values":[[1,""]]}]}]}
```

and

```
$ curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE foo" &&   \
  curl -d '{"database":"foo", "points": [{"measurement": "cpu","fields": {"value": 100}, "tags": {"":"tag_value_with_empty_key"} }]}' -H "Content-Type: application/json" http://localhost:8086/write \
  curl -G http://localhost:8086/query --data-urlencode "q=SHOW SERIES" --data-urlencode "db=foo"
{"results":[{}]}{"results":[{"series":[{"name":"cpu","columns":["_id",""],"values":[[1,"tag_value_with_empty_key"]]}]}]}
```

and 

```
$ curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE foo" &&  \
  curl -d '{"database":"foo", "points": [{"measurement": "cpu","fields": {"value": 100}, "tags": {"":""} }]}' -H "Content-Type: application/json" http://localhost:8086/write && \
  curl -G http://localhost:8086/query --data-urlencode "q=SHOW SERIES" --data-urlencode "db=foo"
{"results":[{}]}{"results":[{"series":[{"name":"cpu","columns":["_id",""],"values":[[1,""]]}]}]}
```

### PR DESCRIPTION
**Pull request by jwilder**
Some combinations of tag and fields would get parsed correctly when they should not.

Fixes #2678 #3061

### CODE REVIEW

File: tsdb/points.go [hunk 1]
```diff
@@ -205,25 +205,40 @@ func scanKey(buf []byte, i int) (int, []byte, error) {
 	// tracks how many commas we've seen so we know how many values are indices.
 	// Since indices is an arbitraily large slice,
 	// we need to know how many values in the buffer are in use.
-	separators := 0
+	commas := 0
```
Comment (otoolep) [inline tsdb/points.go]: Minor: what about `nEquals` and `nCommas`?

**Comment by otoolep** [comment]
+1.

**Comment by otoolep** [comment]
Don't forget the cherry pick.

File: tsdb/points_test.go [hunk 2]
```diff
@@ -594,7 +662,7 @@ func TestParsePointEscapedStringsAndCommas(t *testing.T) {
 }
 
 func TestParsePointWithStringWithEquals(t *testing.T) {
-	test(t, `cpu,host=serverA,region=us-east str="foo=bar",value=1.0, 1000000000`,
```
Comment (otoolep) [inline tsdb/points_test.go]: What happened here? Was it a typo, that new code now threw an error on?

Comment (jwilder) [inline tsdb/points_test.go]: This test actually shouldn't have passed but was before.  There is a comma at the end of field section.

Comment (otoolep) [inline tsdb/points_test.go]: Yeah, that's what I guessed.

File: tsdb/points.go (hunk 1 above)
Comment (jwilder) [inline tsdb/points.go]: Hmm.  I don't think that adds much in this context.

Comment (otoolep) [inline tsdb/points.go]: Fine with me, purely a style question.

**Pull request merged**